### PR TITLE
Change .metadata.name to optional for all Patch resources

### PR DESCRIPTION
**Pull request by lblackstone**
### Proposed changes
This is a minor breaking change that updates the `.metadata.name` field from required to optional for all Patch resources. This resolves an issue where a nested reference to the ObjectMeta type also required the name field to be set (e.g., for Deployment, StatefulSet, etc.). Instead of requiring the name field in the SDKs, this requirement will be enforced by the provider. This change trades some clarity in the API usage for semantic correctness.

### Related issues (optional)
Fix https://github.com/pulumi/pulumi-kubernetes/issues/2115

### CODE REVIEW

**Comment by viveklak** [comment]
> This is a minor breaking change
I am not sure this is a breaking change since existing code should still work properly?

> this requirement will be enforced by the provider
Should we roll that into this PR as well?

**Comment by lblackstone** [comment]
> I am not sure this is a breaking change since existing code should still work properly?

It changes types, which affects Go, C#, and Java. I'd consider this a minor break still since the feature is opt-in.

File: provider/pkg/provider/provider.go [hunk 1]
```diff
@@ -1262,6 +1262,12 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 	oldInputs := propMapToUnstructured(olds)
 	newInputs := propMapToUnstructured(news)
 
+	if k.serverSideApplyMode && strings.HasSuffix(urn.Type().String(), "Patch") {
```
Comment (viveklak) [inline provider/pkg/provider/provider.go:1265]: Nit: would be good to pull `strings.HasSuffix(urn.Type().String(), "Patch")` to a function like `isPatchResource()` or something.

**Review by viveklak — APPROVED** [review]

File: provider/pkg/provider/provider.go (hunk 1 above)
Comment (lblackstone) [reply provider/pkg/provider/provider.go:1265]: Agreed -- I'll follow up on that in a later change.

**Comment by viveklak** [comment]
> 

I think in general compilation breaks are ok as long as replacements etc. arent triggered. As you said, the point is moot here since its in preview.

**Pull request merged**